### PR TITLE
SIMSBIOHUB-567: Add/Support New Observation Environment Attributes

### PR DESCRIPTION
**Pull request by LouisThedroux**
## Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-567

## Description of Changes

- Add new UI to support environment columns in observations table.
- Add new db tables for environment attribute codes.
- Inserted environmental variables and options as a new migration. 

## Testing Notes

Can add/remove environment attribute columns in observations table.
Can add/remove environment column cell values.
Can import observation csv with environment columns.

### CODE REVIEW

File: database/src/migrations/20240502130000_env_variables_create.ts [hunk 1]
```diff
@@ -0,0 +1,2191 @@
...
+),
+(
+  'Temperature Variance',
+  'The water temperature variance in +/- degrees Celsius.',
+  0,
+  50,
+  'celsius'
+),
+(
+  'Sightability Correction Factor.',
```
Comment (LouisThedroux) [inline database/src/migrations/20240502130000_env_variables_create.ts]: Period

File: app/src/features/surveys/observations/observations-table/configure-columns/components/general/ConfigureGeneralColumns.tsx [hunk 2]
```diff
@@ -1,71 +1,74 @@
...
 import FormControlLabel from '@mui/material/FormControlLabel';
-import IconButton from '@mui/material/IconButton';
 import List from '@mui/material/List';
 import ListItem from '@mui/material/ListItem';
 import ListItemButton from '@mui/material/ListItemButton';
 import ListItemIcon from '@mui/material/ListItemIcon';
 import ListItemText from '@mui/material/ListItemText';
 import Stack from '@mui/material/Stack';
 import Typography from '@mui/material/Typography';
+import Box from '@mui/system/Box';
```
Comment (aubertinyoung) [inline app/src/features/surveys/observations/observations-table/configure-columns/components/general/ConfigureGeneralColumns.tsx]: should be ```import Box from '@mui/material/Box';```

**Review by mauberti-bc — DISMISSED** [review]
Tested locally and worked as expected 👍

File: app/src/features/surveys/observations/observations-table/configure-columns/components/measurements/useConfigureMeasurementColumns.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      // Delete the measurement columns from the database
+      observationsTableContext.deleteObservationMeasurementColumns(measurementColumnsToRemove, () => {
+        // Remove the measurement columns from the table context
+        observationsTableContext.setMeasurementColumns((currentColumns) => {
+          const remainingColumns = currentColumns.filter(
+            (currentColumn) => !measurementColumnsToRemove.includes(currentColumn.taxon_measurement_id)
+          );
+
+          // Store all remaining measurement definitions in local storage
+          sessionStorage.setItem(
```
Comment (MacDeluca) [inline app/src/features/surveys/observations/observations-table/configure-columns/components/measurements/useConfigureMeasurementColumns.tsx:65]: Once 572 goes in, you can use `usePersistentState` hook if you want

Comment (NickPhura) [reply app/src/features/surveys/observations/observations-table/configure-columns/components/measurements/useConfigureMeasurementColumns.tsx:65]: Ill look at incorporating it in a separate PR, just to get his chunk in sooner.

I did start to drop it in, but in a few places I do some filtering of the state (which could maybe be removed, but not sure), that will require a bit more testing to make sure I don't accidentally break anything.

**Comment by MacDeluca** [comment]
Code looks great
Few things I noticed while testing the app, you can omit them if they are intended or beyond the scope of this PR.

1. Nit: After refreshing the page the 'SHOW/HIDE ALL' button only works on second click
2. Hiding all the columns and refreshing the page doesnt keep the state (intended?)
3. Clicking configure and searching for 'antler configuration' -> then clicking Add Record crashes the UI with error: ```Cannot read properties of undefined (reading 'value')```. I think this is happening for both measurements and environment columns
4. Nit: The measurements in the dropdown could be capitalized similar to the environment options
5. Downloading CSV template and importing data throws error for this specific CSV. I very possibly could have formatted the data incorrectly.
[SIMS_Observations_Template_2024_05_22_10_28_07.csv](https://github.com/bcgov/biohubbc/files/15406897/SIMS_Observations_Template_2024_05_22_10_28_07.csv)
image

**Comment by NickPhura** [comment]
@MacQSL 

Pushed up a commit with some fixes. I believe I've addressed everything you mentioned.

**Review by MacDeluca — APPROVED** [review]
LGTM 😺

**Pull request merged**